### DISCUSSION TITLE
MS SQL Tables

### DISCUSSION THREAD

**Discussion by km37500** [directus/directus#12348] [Archive]
Hi 

I have successfully installed the current directus version with ms sql 2016 and there are no problems to login, create and edit new tables.

But if I want to edit the existing tables I always got the error : [FORBIDDEN] You don't have permission to access this.

How can I find more details to the sql error here?

Even with a simple test table in MS SQL:

```
CREATE TABLE [dbo].[test](
	[id] [int] IDENTITY(1,1) NOT NULL,
	[name] [varchar](50) NULL
) ON [PRIMARY]
GO
```

I am not able to edit/import/configure in directus.

In mysql I have no problems to import existing tables.

In ms sql:

What are the requirements of the table / field structure so that directus is able to edit / configure it?

**Comment by km37500** [comment]
I have tested some more table, some will work (the fields are visible in data model), some not (all fields are not visible in directus data model). 

I guess there are some minimal requirements for the tables. is this correct?

**Comment by rijkvanzanten** [comment]
Do those tables have a primary key? Directus requires tables to have primary keys in order to use them with Directus. This should also show up as warnings in stdout

**Comment by km37500** [comment]
Thanks for the advice. With autoincrement primary keys it worked.
